125 ticket component (#128)
ministryofjustice/moj-frontend: src/moj/components/ticket-panel/README.md

> # Timeline

Author: @DanGuyMOJ are you able to transfer the context and research of this component here. Things like usage and use when sort of content

- [Guidance](https://moj-design-system.herokuapp.com/components/ticket-panel)
- [Preview](https://moj-frontend.herokuapp.com/components/ticket-panel)

### Installation

You will need to install the following code at the bottom of `server.js`, just above `module.exports = app;`

```
// Add filters from MOJ Frontend
let mojFilters = require('./node_modules/@ministryofjustice/frontend/filters/all')();
mojFilters = Object.assign(mojFilters);
Object.keys(mojFilters).forEach(function (filterName) {
  nunjucksAppEnv.addFilter(filterName, mojFilters[filterName])
});
```

## Example
Below is a typical example of the timeline component in use.

> ```

Author: @johnnolan update this with actual example of usage that has now been confirmed

# Ticket Panel

- [Guidance](https://moj-design-system.herokuapp.com/components/ticket-panel)
- [Preview](https://moj-frontend.herokuapp.com/components/ticket-panel)

## Example
Below is a typical example of the timeline component in use.

```
{{ mojTicketPanel({

>         ariaLabel: 'Sub navigation 1',
>         items: [{
>           html: ' <h2 class="govuk-heading-m govuk-!-margin-bottom-2">This is a heading 2</h2>
>           <p class="govuk-body">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
>           <a class="govuk-button govuk-!-margin-bottom-1" data-module="govuk-button">
>             Save and continue
>           </a>',
>           ariaLabel: 'Section 1'
>         }]

Reviewer: ```suggestion
        attributes: {
          'aria-label': 'Sub navigation 1',
        },
        items: [{
          html: ' <h2 class="govuk-heading-m govuk-!-margin-bottom-2">This is a heading 2</h2>
          <p class="govuk-body">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
          <a class="govuk-button govuk-!-margin-bottom-1" data-module="govuk-button">
            Save and continue
          </a>',
          attributes: {
            'aria-label': 'Section 1'
          }
        }]
```

      }) }}
```

## Arguments

This component accepts the following arguments.

### Container

|Name|Type|Required|Description|
|---|---|---|---|
|classes|string|No|Classes to add to the timeline's container.|
|attributes|object|No|HTML attributes (for example data attributes) to add to the timeline's container.|

> |ariaLabel|string|No|Adds a aria-label to the component describing its context|

Reviewer: ```suggestion
```

|inline|string|No|Changes the default view from column to row (stacked)|

### Items

|Name|Type|Required|Description|
|---|---|---|---|

> |colour|object|No|Sets the side bad colour of the component.|

Reviewer: ```suggestion
|classes|string|No|Classes to add to the component, such as `moj-ticket-panel__content--red`|
```
